repl: don't leak CLI arguments

matklad opened the pull request:
The first commit corrects a couple of minor error with repl's arguments. The two following comments refacro related code somewhat to hopefully make memory management somewhat easier.

matklad committed:
repl: don't leak CLI arguments

- `addresses` is allocated and must be freed
- `args_allocated` must be freed by someone ...
- but also repl stores `command`, which points inside `args_allocated`,
  so it should keep those alive.

matklad committed:
cli: simplify memory management

Let's make it the caller's problem/privilege to define the lifetime of
cli args here, so that's only one `init/deinit` pair syntactically.

matklad committed:
flags: centralize argv[0] skipping logic

kprotty commented on src/flags.zig:
```diff
@@ -72,6 +72,8 @@ pub fn fatal(comptime fmt_string: []const u8, args: anytype) noreturn {
 ///
 /// If `pub const help` declaration is present, it is used to implement `-h/--help` argument.
 pub fn parse(args: *std.process.ArgIterator, comptime CliArgs: type) CliArgs {
+    assert(args.skip()); // Discard executable name.
```
Should the `pub fn main()` in this file also go through `parse()` (it duplicates the skip logic).

matklad replied:
Excellent catch @kprotty! Looks like we were skipping these tests! Should be fixed in the latest commit.

matklad committed:
flags: revive tests

Looks like upgrading Zig broke `ZIG_EXE` logic. Haven't traced this to
the source yet, but looks like this var isn't passed when using test
runner in server mode perhaps?

matklad commented:
Upstream issue: https://github.com/ziglang/zig/issues/18672
